### PR TITLE
Unit vs units change in data metaschema

### PR DESCRIPTION
**Pull request by GernotMaier**
This pull requests fixes a couple of additional items related to names changes / new simulation model outline in PR #761:

- change of `units` to `unit` for the database keys requires also to change this in the metaschema.
- (related) change of data model urls (schema links) to [ctao gitlab repo](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters/-/tree/main/schema?ref_type=heads)
- update of instrument types in data.metaschema for new telescope naming (not ideal: list of allowed array elements is both in the metaschema and in names.py.

## TODO - fix before merging:

- [ ] tests/resources/MST_mirror_2f_measurements.schema.yml includes a temporary link to this branch (otherwise unit test would fail.

### CODE REVIEW

File: simtools/utils/general.py [hunk 1]
```diff
@@ -917,12 +917,13 @@ def get_value_unit_type(value, unit_str=None):
...
-                base_value = base_value * u.Unit(base_unit).to(u.Unit(unit_str))
-            except u.UnitConversionError:
-                _logger.error(f"Cannot convert {base_unit} to {unit_str}.")
-                raise
+        try:
+            base_value = base_value * u.Unit(base_unit).to(u.Unit(unit_str))
+        except u.UnitConversionError:
+            _logger.error(f"Cannot convert {base_unit} to {unit_str}.")
+            raise
+        except TypeError:
```
Comment (VictorBarbosaMartins) [inline simtools/utils/general.py:925]: it is not clear to me when the TypeError occurs? If base_unit is None?

We are still not checking whether unit_str makes sense before proceeding. Passing None or "" to `unit_str` leads to None being returned as base_unit.

File: simtools/utils/general.py [hunk 2]
```diff
@@ -917,12 +917,13 @@ def get_value_unit_type(value, unit_str=None):
         base_type = extract_type_of_value(base_value)
 
     if unit_str is not None:
-        if base_unit is not None:
```
Comment (VictorBarbosaMartins) [inline simtools/utils/general.py:920]: Pycharm complains that lines 902 and 904 are not needed (`base_value = value`, `base_type = ""`) as these arguments are always defined below. I suggest removing them

**Review by VictorBarbosaMartins — COMMENTED** [review]
Hi @GernotMaier, I leave a couple of comments here, but I prefer to wait for Orel to look into the test before approving it. Thanks!

File: simtools/utils/general.py (hunk 1 above)
Comment (GernotMaier) [reply simtools/utils/general.py:925]: Exactly, u.Unit(Unit) gives you a TypeError.

And yes, returning None is also ok for when there is no base unit. This is intended (also in the unit tests)

File: tests/unit_tests/db/test_db_handler.py [hunk 3]
```diff
@@ -287,6 +287,37 @@ def test_adding_new_parameter_db(db, random_id, db_cleanup, io_handler):
     assert pars["new_test_parameter_quantity_str"]["type"] == "float"
     assert pars["new_test_parameter_quantity_str"]["unit"] == "cm"
 
+    # site parameters
+    db.add_new_parameter(
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply tests/unit_tests/db/test_db_handler.py:291]: Yes, this is fine - no worries.

This is using a sandbox database, similar to the existing tests for adding parameters. I am just adding two more cases.

**Comment by GernotMaier** [comment]
@VictorBarbosaMartins - thanks for checking this PR. Please have another look and let me know if this is ok.

File: tests/unit_tests/db/test_db_handler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (VictorBarbosaMartins) [reply tests/unit_tests/db/test_db_handler.py:291]: ok

File: simtools/utils/general.py (hunk 2 above)
Comment (VictorBarbosaMartins) [reply simtools/utils/general.py:920]: I still see that we can remove `base_type = ""`. i am trying to reproduce locally and see which error you are talking about, but tests are just taking too long

Comment (VictorBarbosaMartins) [reply simtools/utils/general.py:920]: If you look carefully, one can see that `base_type` is defined in the function regardless of the outcome of the try and except statement and the if statement, so this first declaration of `base_type=""` should really be removed. Too bad I cant look at it in a reasonable amount of test time (I didnt see a problem when testing general.py alone )

**Comment by VictorBarbosaMartins** [comment]
I think we should include the URL model to the .env_template file as well

**Comment by VictorBarbosaMartins** [comment]
Perhaps you also want to have a look when the env variable is defined based on the cloned repository. I have this line in my .env file and with that I dont get the unit tests to work:
`export SIMTOOLS_DB_SIMULATION_MODEL_URL="../model_parameters/"`
where in model_parameters I cloned with `git clone https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters`. The errors are the following:

```
========================= 27 failed, 295 passed, 11 skipped, 73 errors in 124.28s (0:02:04) =========================
```

All of them are similar to

```
ERROR tests/unit_tests/simtel/test_simtel_runner_ray_tracing.py::test_load_required_files - AttributeError: 'NoneType' object has no attribute 'get'
```

**Comment by GernotMaier** [comment]
Why "EXPORT"? It should be "SIMTOOLS_DB_SIMULATION_MODEL_URL="../model_parameters/"

> On 5. Mar 2024, at 15:27, Victor Barbosa Martins ***@***.***> wrote:
> 
> 
> Perhaps you also want to have a look when the env variable is defined based on the cloned repository. I have this line in my .env file and with that I dont get the unit tests to work:
> export SIMTOOLS_DB_SIMULATION_MODEL_URL="../model_parameters/"
> where in model_parameters I cloned with git clone https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters. The errors are the following:
> ========================= 27 failed, 295 passed, 11 skipped, 73 errors in 124.28s (0:02:04) =========================
> 
> All of them are similar to
> ERROR tests/unit_tests/simtel/test_simtel_runner_ray_tracing.py::test_load_required_files - AttributeError: 'NoneType' object has no attribute 'get'
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>



---
Dr Gernot Maier
CTA
Deutsches Elektronen-Synchrotron DESY 
Ein Forschungszentrum der Helmholtz-Gemeinschaft
Platanenallee 6, D-15738 Zeuthen, Germany

Tel.: +49  33 7627 7598
Besucheradresse: 1X07

Working hours differ and mine may not be the same as yours.  Please do not feel obliged to reply to this e-mail outside of your normal working hours.

File: simtools/utils/general.py (hunk 2 above)
Comment (GernotMaier) [reply simtools/utils/general.py:920]: OK. Fixed.

**Comment by GernotMaier** [comment]
> I think we should include the URL pointing to model parameters to the .env_template file as well

My plan is to get those missing parameters into the database before merging the whole new simulation model setup to main. So this will not be necessary when we have updated the database and it is a temporary setup.

**Review by VictorBarbosaMartins — APPROVED** [review]

**Pull request merged**